

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="es" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="es" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Introducción &mdash; documentación de Call me by your reads - 0.0.1</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
        <script src="_static/clipboard.min.js"></script>
        <script src="_static/copybutton.js"></script>
        <script src="_static/translations.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/copybutton.css" type="text/css" />
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
    <link rel="index" title="Índice" href="genindex.html" />
    <link rel="search" title="Búsqueda" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Call me by your reads
          

          
          </a>

          
            
            
              <div class="version">
                0.0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Introducción</a></li>
<li><a class="reference internal" href="#instalacion-de-gatk">Instalación de GATK</a></li>
<li><a class="reference internal" href="#moviendole-a-los-alineamientos">Moviendole a los alineamientos</a></li>
<li><a class="reference internal" href="#llamado-de-variantes-finally">Llamado de variantes, finally (?)</a></li>
<li><a class="reference internal" href="#analisis-preliminar-de-mis-variantes">Análisis preliminar de mis variantes</a></li>
<li><a class="reference internal" href="#seleccion-de-variantes">Selección de variantes</a></li>
<li><a class="reference internal" href="#formatos-formatos-everywhere">Formatos… formatos everywhere</a></li>
</ul>
</div>
            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Call me by your reads</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Introducción</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/gatk.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="introduccion">
<h1>Introducción<a class="headerlink" href="#introduccion" title="Enlazar permanentemente con este título">¶</a></h1>
<p>Ya tenemos nuestras lecturas alineadas, ya nos quitamos las regiones problemáticas. Es hora de llamar variantes!</p>
</div>
<div class="section" id="instalacion-de-gatk">
<h1>Instalación de GATK<a class="headerlink" href="#instalacion-de-gatk" title="Enlazar permanentemente con este título">¶</a></h1>
<ol class="arabic simple">
<li><p>Vamos a nuestro directorio <code class="code docutils literal notranslate"><span class="pre">$HOME</span></code>, as usual</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ cd $HOME
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Bajamos la versión actual de GATK4</p></li>
</ol>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>Algunos comandos cambian ligeramente entre versiones, de modo que si quieres replicar los comandos de esta documentación, debes bajar la versión 4.1.9.0 de GATK</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ wget https://github.com/broadinstitute/gatk/releases/download/4.1.9.0/gatk-4.1.9.0.zip
</pre></div>
</div>
</div>
<ol class="arabic simple" start="3">
<li><p>Descomprimimos</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ unzip gatk-4.1.9.0.zip
</pre></div>
</div>
<ol class="arabic simple" start="4">
<li><p>Agregamos la carpeta de GATK a nuestro PATH</p></li>
</ol>
<div class="admonition danger">
<p class="admonition-title">Peligro</p>
<p>Recuerda que los cambios al PATH son potencialmente peligrosos, estos comandos debes ejecutarlos una sola vez</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ echo &quot;PATH=\$HOME/gatk-4.1.9.0:\$PATH&quot; &gt;&gt; $HOME/.bashrc
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ echo &quot;export PATH&quot; &gt;&gt; $HOME/.bashrc
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ echo &quot;source \$HOME/gatk-4.1.9.0/gatk-completion.sh&quot; &gt;&gt; $HOME/.bashrc
</pre></div>
</div>
</div>
<ol class="arabic simple" start="5">
<li><p>Reiniciamos sesión en la instancia</p></li>
</ol>
</div>
<div class="section" id="moviendole-a-los-alineamientos">
<h1>Moviendole a los alineamientos<a class="headerlink" href="#moviendole-a-los-alineamientos" title="Enlazar permanentemente con este título">¶</a></h1>
<p>Antes de realizar el llamado de variantes, debemos hacer algunos pasos de recalibración, esto es importante ya que en aplicaciones de salud debemos ser capaces de garantizar que los hallazgos son de buena calidad y que pueden ser accionables.</p>
<ol class="arabic simple">
<li><p>Preparamos el escenario</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ cd $HOME
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ ln -s /usr/local/bioinformatics/bundle .
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mkdir -p $HOME/04_gatk
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ cd $HOME/04_gatk
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Necesitamos el archivo de coordenadas para que GATK sepa donde buscar</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ cp $HOME/00_other/TSO_xt_hg38.bed .
</pre></div>
</div>
<ol class="arabic simple" start="3">
<li><p>Necesitamos el archivo de alineamiento de lecturas y también su índice</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ ln -s $HOME/03_bwa/S3.bam .
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ ln -s $HOME/03_bwa/S3.bam.bai .
</pre></div>
</div>
<ol class="arabic simple" start="4">
<li><p>Primero agregamos un <em>readgroup</em> a nuestro archivo BAM para que en este archivo quede explicito:</p></li>
</ol>
<ul class="simple">
<li><p>El identificador del <em>readgroup</em></p></li>
<li><p>El nombre de la biblioteca</p></li>
<li><p>La plataforma de secuenciación</p></li>
<li><p>El nombre de la muestra</p></li>
<li><p>La unidad de la plataforma (barcodes)</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk AddOrReplaceReadGroups \
  --INPUT S3.bam \
  --OUTPUT S3.rg.bam \
  --RGLB S3 \
  --RGID S3 \
  --RGPL Illumina \
  --RGSM S3 \
  --RGPU S3
</pre></div>
</div>
<div class="toggle admonition">
<p class="admonition-title">Readgroups?</p>
<blockquote>
<div><p>«There is no formal definition of what a “read group” is, however in practice this term refers to a set of reads that are generated from a single run of a sequencing instrument.»</p>
<p>Aquí te dejamos un <a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups">link</a> con más información.</p>
</div></blockquote>
</div>
<ol class="arabic simple" start="5">
<li><p>Ya que tenemos nuestro bam con readgroup, volvemos a ordenar nuestras lecturas</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk SortSam \
  --INPUT S3.rg.bam \
  --OUTPUT S3.sorted.bam \
  --SORT_ORDER coordinate
</pre></div>
</div>
<ol class="arabic simple" start="6">
<li><p>La importancia de ser único y diferente</p></li>
</ol>
<a class="reference internal image-reference" href="_images/dedup.png"><img alt="_images/dedup.png" src="_images/dedup.png" style="width: 600px;" /></a>
<p>Con el siguiente comando podemos eliminar duplicados ópticos o de PCR</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk MarkDuplicates \
  --INPUT S3.sorted.bam \
  --OUTPUT S3.dupmarked.bam \
  --METRICS_FILE S3.dupmarked.txt \
  --CREATE_INDEX true
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Advertencia</p>
<p>En paneles de secuenciación dirigida, en análisis de genoma mitocondrial, en búsqueda de variantes de número de copia, y en GBS; el paso de desduplicación no se recomienda!</p>
</div>
<ol class="arabic simple" start="7">
<li><p>Si trabajamos con organismos módelo, si tenemos estudios previos de llamado de variantes, podemos mejorar muchísimo el desempeño de los programas indicándoles el camino a seguir</p></li>
</ol>
<p>7.1 Análisis de la distribución de scores de calidad</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk BaseRecalibrator \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --input S3.dupmarked.bam \
  --known-sites $HOME/bundle/Homo_sapiens_assembly38.dbsnp138.vcf \
  --known-sites $HOME/bundle/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz \
  --output S3_recal-data.table \
  --intervals TSO_xt_hg38.bed
</pre></div>
</div>
<p>7.2 Recalibración de los scores de calidad</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk ApplyBQSR \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --input S3.dupmarked.bam \
  --bqsr-recal-file S3_recal-data.table \
  --output S3_recal-reads.bam \
  --intervals TSO_xt_hg38.bed
</pre></div>
</div>
<p>7.3 Análisis de la distribución de scores de calidad post recalibración</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk BaseRecalibrator \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --input S3_recal-reads.bam \
  --known-sites $HOME/bundle/Homo_sapiens_assembly38.dbsnp138.vcf \
  --known-sites $HOME/bundle/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz \
  --output S3_post-recal-data.table \
  --intervals TSO_xt_hg38.bed
</pre></div>
</div>
<p>7.4 Verificación del efecto de la recalibración</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk AnalyzeCovariates \
  --before-report-file S3_recal-data.table \
  --after-report-file S3_post-recal-data.table \
  --plots-report-file S3_recal-plots.pdf
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Advertencia</p>
<p>Si no tienes datos previos, si tu reacción de secuenciación salio de libro de texto, si trabajas con organismos no módelo, este paso es omisible</p>
</div>
</div>
<div class="section" id="llamado-de-variantes-finally">
<h1>Llamado de variantes, finally (?)<a class="headerlink" href="#llamado-de-variantes-finally" title="Enlazar permanentemente con este título">¶</a></h1>
<p>Una vez que tenemos las lecturas listas, recalibradas, marcadas, desduplicadas, etc. Podemos ahora si llamar las variantes</p>
<ol class="arabic simple">
<li><p>Obtención de variantes crudas</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk HaplotypeCaller \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --input S3_recal-reads.bam \
  --intervals TSO_xt_hg38.bed \
  --stand-call-conf 10.0 \
  --output S3_raw-vars.vcf
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Si trabajamos con organismos módelo, si tenemos estudios previos de llamado de variantes, podemos mejorar muchísimo el desempeño de los programas indicándoles el camino a seguir</p></li>
</ol>
<p>2.1 Análisis de la distribución de scores de calidad: SNVs</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk VariantRecalibrator \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --variant S3_raw-vars.vcf \
  --intervals TSO_xt_hg38.bed \
  --resource:hapmap,known=false,training=true,truth=true,prior=15.0 $HOME/bundle/hapmap_3.3.hg38.vcf.gz \
  --resource:omni,known=false,training=true,truth=true,prior=12.0 $HOME/bundle/1000G_omni2.5.hg38.vcf.gz \
  --resource:1000G,known=false,training=true,truth=false,prior=10.0 $HOME/bundle/1000G_phase1.snps.high_confidence.hg38.vcf.gz \
  --resource:dbsnp,known=true,training=false,truth=false,prior=2.0 $HOME/bundle/Homo_sapiens_assembly38.dbsnp138.vcf \
  --use-annotation QD \
  --use-annotation FS \
  --use-annotation SOR \
  --use-annotation MQ \
  --use-annotation MQRankSum \
  --use-annotation ReadPosRankSum \
  --mode SNP \
  --truth-sensitivity-tranche 100.0 \
  --truth-sensitivity-tranche 99.9 \
  --truth-sensitivity-tranche 99.0 \
  --truth-sensitivity-tranche 90.0 \
  --max-gaussians 1 \
  --max-negative-gaussians 1 \
  --output S3_recalibrate-SNP.recal \
  --tranches-file S3_recalibrate-SNP.tranches \
  --rscript-file S3_recalibrate-SNP-plots.R
</pre></div>
</div>
<p>2.2 Recalibración de variantes: SNVs</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk ApplyVQSR \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --variant S3_raw-vars.vcf \
  --intervals TSO_xt_hg38.bed \
  --mode SNP \
  --truth-sensitivity-filter-level 99.0 \
  --recal-file S3_recalibrate-SNP.recal \
  --tranches-file S3_recalibrate-SNP.tranches \
  --output S3_recal-snps_raw-indels.vcf
</pre></div>
</div>
<p>2.3 Análisis de la distribución de scores de calidad: InDels</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk VariantRecalibrator \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --variant S3_recal-snps_raw-indels.vcf \
  --intervals TSO_xt_hg38.bed \
  --resource:mills,known=true,training=true,truth=true,prior=12.0 $HOME/bundle/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz \
  --use-annotation QD \
  --use-annotation FS \
  --use-annotation SOR \
  --use-annotation MQRankSum \
  --use-annotation ReadPosRankSum \
  --mode INDEL \
  --truth-sensitivity-tranche 100.0 \
  --truth-sensitivity-tranche 99.9 \
  --truth-sensitivity-tranche 99.0 \
  --truth-sensitivity-tranche 90.0 \
  --max-gaussians 1 \
  --max-negative-gaussians 1 \
  --output S3_recalibrate-INDEL.recal \
  --tranches-file S3_recalibrate-INDEL.tranches \
  --rscript-file S3_recalibrate-INDEL-plots.R
</pre></div>
</div>
<p>2.4 Recalibración de variantes: InDels</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk ApplyVQSR \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --variant S3_recal-snps_raw-indels.vcf \
  --intervals TSO_xt_hg38.bed \
  --mode INDEL \
  --truth-sensitivity-filter-level 99.0 \
  --recal-file S3_recalibrate-INDEL.recal \
  --tranches-file S3_recalibrate-INDEL.tranches \
  --output S3_recalibrated_variants.vcf
</pre></div>
</div>
</div>
<div class="section" id="analisis-preliminar-de-mis-variantes">
<h1>Análisis preliminar de mis variantes<a class="headerlink" href="#analisis-preliminar-de-mis-variantes" title="Enlazar permanentemente con este título">¶</a></h1>
<p>Ahora que tenemos nuestras variantes con los scores de calidad adecuados, qué sigue?
El primer paso es ponerles nombre y apellido a las variantes que encontremos</p>
<ol class="arabic simple">
<li><p>Annotación inicial: GATK + dbSNP</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk VariantAnnotator \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --variant S3_recalibrated_variants.vcf \
  --intervals TSO_xt_hg38.bed \
  --dbsnp $HOME/bundle/Homo_sapiens_assembly38.dbsnp138.vcf \
  --output S3_annotated_variants.vcf \
  --annotation Coverage
</pre></div>
</div>
</div>
<div class="section" id="seleccion-de-variantes">
<h1>Selección de variantes<a class="headerlink" href="#seleccion-de-variantes" title="Enlazar permanentemente con este título">¶</a></h1>
<p>Tradicionalmente cuando eliminamos elementos de un dataset, llamamos a este proceso como «filtrado» ya que eliminamos elementos que cumplan cierto critero.</p>
<p>En el argot de GATK las cosas son distintas, para GATK, el filtrado de las variantes implica unicamente etiquetar los elementos que cumplan tal o cual característica.</p>
<p>Después de este largo camino, tenemos variantes, pero no todas son de buena calidad, a pesar de que empleamos muchos puntos de control.</p>
<ol class="arabic simple">
<li><p>Selección de variantes de buena calidad y de buena profundidad</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ gatk SelectVariants \
  --reference $HOME/bundle/Homo_sapiens_assembly38.fasta \
  --variant S3_annotated_variants.vcf \
  --output S3_annotated_qd_dp_filtered_variants.vcf \
  --selectExpressions &quot;QD &gt; 5.0 &amp;&amp; DP &gt; 10.0&quot;
</pre></div>
</div>
</div>
<div class="section" id="formatos-formatos-everywhere">
<h1>Formatos… formatos everywhere<a class="headerlink" href="#formatos-formatos-everywhere" title="Enlazar permanentemente con este título">¶</a></h1>
<div class="toggle admonition">
<p class="admonition-title">BAM</p>
<p>El formato BAM es similar en estructura al formato SAM, sin embargo, es el formato de elección para el manejo de alineamientos ya que está comprimido y ahorra mucho espacio</p>
</div>
<div class="toggle admonition">
<p class="admonition-title">BED</p>
<p>El formato BED consta de archivos de texto plano con estructura tabular que indica posiciones dentro de un genoma</p>
<p>Este formato es indispensable para el manejo óptimo de alineamientos ya que nos permite filtrar unicamente por las regiones que nos interesan!</p>
<p>Hay distintas versiones del formato <a class="reference external" href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a>, la estructura más básica consta de 4 a 6 columnas:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">13</span>      <span class="mi">32316459</span>        <span class="mi">32316526</span>        <span class="n">BRCA2</span>   <span class="o">.</span>       <span class="o">+</span>
<span class="mi">13</span>      <span class="mi">32319075</span>        <span class="mi">32319324</span>        <span class="n">BRCA2</span>   <span class="o">.</span>       <span class="o">+</span>
<span class="mi">13</span>      <span class="mi">32325074</span>        <span class="mi">32325183</span>        <span class="n">BRCA2</span>   <span class="o">.</span>       <span class="o">+</span>
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>El formato BED es 0-based para la segunda columna!</p>
<p>Si mi gen inicia en el nucleótido 345 y termina en el 678 del cromosoma MT, su nomenclatura en formato BED será la siguiente:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">MT</span>      <span class="mi">344</span>     <span class="mi">678</span>     <span class="n">mi_gen</span>  <span class="o">.</span>       <span class="o">+</span>
</pre></div>
</div>
</div>
</div>
<div class="toggle admonition">
<p class="admonition-title">VCF</p>
<p>El formato VCF también es un archivo de texto plano con una estructura tabular:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  g204</span>
<span class="mi">1</span>       <span class="mi">13417</span>   <span class="n">rs777038595</span>     <span class="n">C</span>       <span class="n">CGAGA</span>   <span class="mf">324.73</span>  <span class="n">PASS</span>    <span class="n">AC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">AF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">AN</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">BaseQRankSum</span><span class="o">=</span><span class="mf">0.583</span><span class="p">;</span><span class="n">DB</span><span class="p">;</span><span class="n">DP</span><span class="o">=</span><span class="mi">26</span><span class="p">;</span><span class="n">ExcessHet</span><span class="o">=</span><span class="mf">3.0103</span><span class="p">;</span><span class="n">FS</span><span class="o">=</span><span class="mf">0.000</span><span class="p">;</span><span class="n">MLEAC</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">MLEAF</span><span class="o">=</span><span class="mf">0.500</span><span class="p">;</span><span class="n">MQ</span><span class="o">=</span><span class="mf">22.21</span><span class="p">;</span><span class="n">MQRankSum</span><span class="o">=-</span><span class="mf">0.032</span><span class="p">;</span><span class="n">QD</span><span class="o">=</span><span class="mf">12.49</span><span class="p">;</span><span class="n">ReadPosRankSum</span><span class="o">=-</span><span class="mf">2.185</span><span class="p">;</span><span class="n">SOR</span><span class="o">=</span><span class="mf">0.446</span><span class="p">;</span><span class="n">VQSLOD</span><span class="o">=</span><span class="mf">0.698</span><span class="p">;</span><span class="n">culprit</span><span class="o">=</span><span class="n">ReadPosRankSum</span> <span class="n">GT</span><span class="p">:</span><span class="n">AD</span><span class="p">:</span><span class="n">DP</span><span class="p">:</span><span class="n">GQ</span><span class="p">:</span><span class="n">PL</span>  <span class="mi">0</span><span class="o">/</span><span class="mi">1</span><span class="p">:</span><span class="mi">15</span><span class="p">,</span><span class="mi">11</span><span class="p">:</span><span class="mi">26</span><span class="p">:</span><span class="mi">99</span><span class="p">:</span><span class="mi">362</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">634</span>
</pre></div>
</div>
<div class="toggle admonition">
<p class="admonition-title">Descripción extendida del formato vcf</p>
<blockquote>
<div><table class="docutils align-default">
<colgroup>
<col style="width: 4%" />
<col style="width: 11%" />
<col style="width: 85%" />
</colgroup>
<tbody>
<tr class="row-odd"><td rowspan="2"><p>1: CHROM</p></td>
<td rowspan="2"><p>1</p></td>
<td rowspan="2"><p>Cromosoma en dónde se ubica la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>2: POS</p></td>
<td rowspan="2"><p>1053827</p></td>
<td rowspan="2"><p>Posición de la variante en el cromosoma</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>3: ID</p></td>
<td rowspan="2"><p>rs74685771</p></td>
<td rowspan="2"><p>Identificador de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>4: REF</p></td>
<td rowspan="2"><p>G</p></td>
<td rowspan="2"><p>Alelo de referencia</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>5: ALT</p></td>
<td rowspan="2"><p>C</p></td>
<td rowspan="2"><p>Alelo encontrado en la muestra</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>6: QUAL</p></td>
<td rowspan="2"><p>856.77</p></td>
<td rowspan="2"><p>Score de calidad de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>7: FILTER</p></td>
<td rowspan="2"><p>PASS</p></td>
<td rowspan="2"><p>Flag de calidad de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="34"><p>8:INFO</p></td>
<td rowspan="2"><p>AC=1;</p></td>
<td rowspan="2"><p>Número de alelos alternos encontrados</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>AF=0.500;</p></td>
<td rowspan="2"><p>Frecuencia alelica (alelo alterno)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>AN=2;</p></td>
<td rowspan="2"><p>Número total de alelos para una variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>BaseQRankSum=-2.129</p></td>
<td rowspan="2"><p>Z-score de la calidad de las bases mapeadas en el alelo alterno vs el alelo de referencia empleando una prueba de Wilcoxon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ClippingRankSum=-0.286;</p></td>
<td rowspan="2"><p>Z-score del número de lecturas con calidad de mapeo bajas para el alelo alterno vs el alelo de referencia empleando una prueba de Wilcoxon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>DB;</p></td>
<td rowspan="2"><p>Indica si la variante pertenece a una base de datos (dbSNP)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>DP=63;</p></td>
<td rowspan="2"><p>número de lecturas mapeadas en la posición de la variante</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>FS=0;</p></td>
<td rowspan="2"><p>P-value ajustado para determinar sesgo hacía una cadena de DNA empleando una prueba exacta de Fisher</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MLEAC=1;</p></td>
<td rowspan="2"><p>Máxima verosimilitud esperada para el número de alelos observados (depende del número de muestras).</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MLEAF=0.500;</p></td>
<td rowspan="2"><p>Máxima verosimilitud esperada para las frecuencias alélicas (depende del número de muestras).</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MQ=60.00;</p></td>
<td rowspan="2"><p>Calidad media de mapeo en la posición de la variante.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>MQRankSum=-0.0.635;</p></td>
<td rowspan="2"><p>Z-score de la calidad de mapeo las bases alíneadas en el alelo alterno vs el alelo de referencia empleando una prueba de Wilcoxon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>QD=13.60;</p></td>
<td rowspan="2"><p>Confianza del alelo observado (Quality over depth)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ReadPosRankSum=0.621;</p></td>
<td rowspan="2"><p>Z-score del sesgo posicional (con respecto de la longitud de la lectura) del alelo alternativo vs el alelo de referencia (depende del número de muestras)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>SOR=0.768;</p></td>
<td rowspan="2"><p>Suma simétrica de la razón de momios del número de lecturas que presentan la variante cerca del final de la lectura, vs las lecturas que la presentan cerca del inicio de la lectura.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>VQSLOD=3.39;</p></td>
<td rowspan="2"><p>Logaritmo de la razón de momios de que la variante sea verdadera vs que sea un falso positivo.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>culprit=qd</p></td>
<td rowspan="2"><p>El descriptor que presentó la métrica más desfavorable</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="10"><p>9:FORMAT</p></td>
<td rowspan="2"><p>GT</p></td>
<td rowspan="2"><p>Genotipo</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>AD</p></td>
<td rowspan="2"><p>Profundidad de los alelos observados (Ref,Alt)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>DP</p></td>
<td rowspan="2"><p>Profundidad total (número de lecturas</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>GQ</p></td>
<td rowspan="2"><p>Calidad del genotipado</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>PL</p></td>
<td rowspan="2"><p>Verosimilitud de genotipos (Obs,Nul,Alt)</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="10"><p>10:g204</p></td>
<td rowspan="2"><p>0/1</p></td>
<td rowspan="2"><p>Heterocigoto</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>37,26</p></td>
<td rowspan="2"><p>(G)37 reads + (C)26 reads</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>63</p></td>
<td rowspan="2"></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>99</p></td>
<td rowspan="2"></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>885,0,1386</p></td>
<td rowspan="2"></td>
</tr>
<tr class="row-even"></tr>
</tbody>
</table>
</div></blockquote>
</div>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>El formato VCF es 1-based en la segunda columna, por lo que es importante considerar esto cuando manejemos combinaciones de formatos (como con bedtools y bcftools)</p>
</div>
</div>
<div class="toggle admonition">
<p class="admonition-title">GFF</p>
<p>El formato GFF consta de archivos de texto plano con estructura tabular que indica posiciones dentro de un genoma, pueden ser genes, mRNAs, exones, regiones codificantes, y un largo etcetera.</p>
<p>Este formato es uno de los estándares de anotación genómica y se emplea en todos los proyectos de genómica como referencia para obtener genes</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">11869</span>   <span class="mi">12227</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00002234944</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00002234944</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">1</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">lnc_RNA</span> <span class="mi">11869</span>   <span class="mi">14409</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">ID</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">Parent</span><span class="o">=</span><span class="n">gene</span><span class="p">:</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">DDX11L1</span><span class="o">-</span><span class="mi">202</span><span class="p">;</span><span class="n">biotype</span><span class="o">=</span><span class="n">processed_transcript</span><span class="p">;</span><span class="n">tag</span><span class="o">=</span><span class="n">basic</span><span class="p">;</span><span class="n">transcript_id</span><span class="o">=</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">transcript_support_level</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">pseudogene</span>      <span class="mi">11869</span>   <span class="mi">14409</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">ID</span><span class="o">=</span><span class="n">gene</span><span class="p">:</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">DDX11L1</span><span class="p">;</span><span class="n">biotype</span><span class="o">=</span><span class="n">transcribed_unprocessed_pseudogene</span><span class="p">;</span><span class="n">description</span><span class="o">=</span><span class="n">DEAD</span><span class="o">/</span><span class="n">H</span><span class="o">-</span><span class="n">box</span> <span class="n">helicase</span> <span class="mi">11</span> <span class="n">like</span> <span class="mi">1</span> <span class="p">[</span><span class="n">Source</span><span class="p">:</span><span class="n">HGNC</span> <span class="n">Symbol</span><span class="o">%</span><span class="mi">3</span><span class="n">BAcc</span><span class="p">:</span><span class="n">HGNC</span><span class="p">:</span><span class="mi">37102</span><span class="p">];</span><span class="n">gene_id</span><span class="o">=</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">logic_name</span><span class="o">=</span><span class="n">havana_homo_sapiens</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">5</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12010</span>   <span class="mi">12057</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001948541</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001948541</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">1</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">1</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">pseudogenic_transcript</span>  <span class="mi">12010</span>   <span class="mi">13670</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">ID</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Parent</span><span class="o">=</span><span class="n">gene</span><span class="p">:</span><span class="n">ENSG00000223972</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">DDX11L1</span><span class="o">-</span><span class="mi">201</span><span class="p">;</span><span class="n">biotype</span><span class="o">=</span><span class="n">transcribed_unprocessed_pseudogene</span><span class="p">;</span><span class="n">tag</span><span class="o">=</span><span class="n">basic</span><span class="p">;</span><span class="n">transcript_id</span><span class="o">=</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">transcript_support_level</span><span class="o">=</span><span class="n">NA</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12179</span>   <span class="mi">12227</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001671638</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001671638</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12613</span>   <span class="mi">12697</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001758273</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001758273</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">3</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12613</span>   <span class="mi">12721</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000456328</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00003582793</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00003582793</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">2</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">1</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">12975</span>   <span class="mi">13052</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001799933</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001799933</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">4</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
<span class="mi">1</span>       <span class="n">havana</span>  <span class="n">exon</span>    <span class="mi">13221</span>   <span class="mi">13374</span>   <span class="o">.</span>       <span class="o">+</span>       <span class="o">.</span>       <span class="n">Parent</span><span class="o">=</span><span class="n">transcript</span><span class="p">:</span><span class="n">ENST00000450305</span><span class="p">;</span><span class="n">Name</span><span class="o">=</span><span class="n">ENSE00001746346</span><span class="p">;</span><span class="n">constitutive</span><span class="o">=</span><span class="mi">0</span><span class="p">;</span><span class="n">ensembl_end_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">ensembl_phase</span><span class="o">=-</span><span class="mi">1</span><span class="p">;</span><span class="n">exon_id</span><span class="o">=</span><span class="n">ENSE00001746346</span><span class="p">;</span><span class="n">rank</span><span class="o">=</span><span class="mi">5</span><span class="p">;</span><span class="n">version</span><span class="o">=</span><span class="mi">2</span>
</pre></div>
</div>
<p>La estructura del formato GFF consta de 9 columnas:</p>
<blockquote>
<div><table class="docutils align-default">
<colgroup>
<col style="width: 26%" />
<col style="width: 74%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head" rowspan="2"><p>Campo</p></th>
<th class="head" rowspan="2"><p>Ejemplo</p></th>
</tr>
<tr class="row-even"></tr>
</thead>
<tbody>
<tr class="row-odd"><td rowspan="2"><p>1: CHROM</p></td>
<td rowspan="2"><p>1</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>2: SOURCE</p></td>
<td rowspan="2"><p>havana</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>3: FEATURE</p></td>
<td rowspan="2"><p>exon</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>4: START</p></td>
<td rowspan="2"><p>13221</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>5: END</p></td>
<td rowspan="2"><p>13374</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>6: SCORE</p></td>
<td rowspan="2"><p>.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>7: STRAND</p></td>
<td rowspan="2"><p>+</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>8:FRAME</p></td>
<td rowspan="2"><p>.</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="16"><p>9:ATTRIBUTE</p></td>
<td rowspan="2"><p>Parent=transcript:ENST00000450305;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>Name=ENSE00001746346;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>constitutive=0;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ensembl_end_phase=-1;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>ensembl_phase=-1;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>exon_id=ENSE00001746346;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>rank=5;</p></td>
</tr>
<tr class="row-even"></tr>
<tr class="row-odd"><td rowspan="2"><p>version=2</p></td>
</tr>
<tr class="row-even"></tr>
</tbody>
</table>
</div></blockquote>
<div class="admonition important">
<p class="admonition-title">Importante</p>
<p>El formato GFF es 1-based para la quinta columna!</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2021, @atgenomics, @vflorelo, @dianolasa, @zorbax

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
  
 <script type="text/javascript">
    $(document).ready(function() {
        $(".toggle > *").hide();
        $(".toggle .admonition-title").show();
        $(".toggle .admonition-title").click(function() {
            $(this).parent().children().not(".admonition-title").toggle(400);
            $(this).parent().children(".admonition-title").toggleClass("open");
        })
    });
</script>


</body>
</html>